components/segmenter/src/lib.rs

```diff
@@ -167,3 +167,14 @@ pub use crate::word::WordBreakIteratorLatin1;
 pub use crate::word::WordBreakIteratorPotentiallyIllFormedUtf8;
 pub use crate::word::WordBreakIteratorUtf16;
 pub use crate::word::WordBreakIteratorUtf8;
+
+pub(crate) mod private {
+    /// Trait marking other traits that are considered unstable and should not generally be
+    /// implemented outside of the segmenter crate.
+    ///
+    /// <div class="stab unstable">
+    /// 🚧 This trait is considered unstable; it may change at any time, in breaking or non-breaking ways,
+    /// including in SemVer minor releases. Do not implement this trait in userland.
+    /// </div>
+    pub trait Sealed {}
```

Comment:
Issue: Not clear to me whether this trait should be hard-sealed. I think unstable-sealed is fine.

Reply:
My preference is not to UnstableSealed unless we have a strong argument for it. The datetime stuff is intended to be extensible, but this is very much using the "type level enum" pattern.